[2024-06-11T17:56:42.771+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-11T17:56:42.790+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_prediction.describe_data manual__2024-06-11T17:56:31.261456+00:00 [queued]>
[2024-06-11T17:56:42.798+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_prediction.describe_data manual__2024-06-11T17:56:31.261456+00:00 [queued]>
[2024-06-11T17:56:42.800+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-06-11T17:56:42.812+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): describe_data> on 2024-06-11 17:56:31.261456+00:00
[2024-06-11T17:56:42.824+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=2424) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-11T17:56:42.827+0000] {standard_task_runner.py:63} INFO - Started process 2442 to run task
[2024-06-11T17:56:42.826+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'stock_prediction', 'describe_data', 'manual__2024-06-11T17:56:31.261456+00:00', '--job-id', '73', '--raw', '--subdir', 'DAGS_FOLDER/demo_***.py', '--cfg-path', '/tmp/tmpwv3ozosh']
[2024-06-11T17:56:42.828+0000] {standard_task_runner.py:91} INFO - Job 73: Subtask describe_data
[2024-06-11T17:56:42.870+0000] {task_command.py:426} INFO - Running <TaskInstance: stock_prediction.describe_data manual__2024-06-11T17:56:31.261456+00:00 [running]> on host ad0cd89ef011
[2024-06-11T17:56:42.953+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Nguyen Khoa Doan' AIRFLOW_CTX_DAG_ID='stock_prediction' AIRFLOW_CTX_TASK_ID='describe_data' AIRFLOW_CTX_EXECUTION_DATE='2024-06-11T17:56:31.261456+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-11T17:56:31.261456+00:00'
[2024-06-11T17:56:42.954+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-11T17:56:42.964+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-11T17:56:42.965+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/demo_airflow.py", line 48, in describe_data
    info = df.info()
           ^^^^^^^
AttributeError: 'NoneType' object has no attribute 'info'
[2024-06-11T17:56:42.981+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_prediction, task_id=describe_data, run_id=manual__2024-06-11T17:56:31.261456+00:00, execution_date=20240611T175631, start_date=20240611T175642, end_date=20240611T175642
[2024-06-11T17:56:42.989+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 73 for task describe_data ('NoneType' object has no attribute 'info'; 2442)
[2024-06-11T17:56:43.001+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-11T17:56:43.020+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-11T17:56:43.022+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
